components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/RequestObjectValidatorImpl.java

```diff
@@ -174,6 +193,70 @@ private boolean checkExpirationTime(RequestObject requestObject) throws RequestO
         return true;
     }
 
+    /**
+     * Validate the request object nbf claim and exp claim according to the FAPI specification.
+     * <a href="https://openid.net/specs/openid-financial-api-part-2-1_0.html#authorization-server">...</a>
+     *
+     * @param requestObject request object
+     * @return true if both claims are valid
+     * @throws RequestObjectException if nbf exp validation fails
+     */
+    protected boolean isValidNbfExp(RequestObject requestObject) throws RequestObjectException {
+
+        Date nbfTime = requestObject.getClaimsSet().getNotBeforeTime();
+        Date expirationTime = requestObject.getClaimsSet().getExpirationTime();
+
+        String errorMsg = null;
+        String errorLog = null;
+        if (nbfTime == null) {
+            errorMsg = "Request Object does not contain Not Before Time.";
+        } else if (expirationTime == null) {
+            errorMsg = "Request Object does not contain Expiration Time.";
+        } else {
+            long timeStampSkewMillis = OAuthServerConfiguration.getInstance().getTimeStampSkewInSeconds() * 1000;
+            long nbfTimeInMillis = nbfTime.getTime();
+            long expirationTimeInMillis = expirationTime.getTime();
+            long currentTimeInMillis = System.currentTimeMillis();
+            // nbf should be older than current time
```

Comment:
use sentence case. fix other comments as well.
```suggestion
            // nbf should be older than current time.
```